v1/ao/internal/config/validators.go

```diff
@@ -26,17 +28,54 @@ func NonDefaultEnv(env string, val string) string {
 	return fmt.Sprintf("env found - %s: \"%s\"", env, val)
 }
 
-const validServiceKeyPattern = `^[a-zA-Z0-9]{64}:\S{1,255}$`
+const (
+	validServiceKeyPattern = `^[a-zA-Z0-9]{64}:.{1,255}$`
+
+	serviceKeyPartsCnt  = 2
+	serviceKeyDelimiter = ":"
+
+	spacesPattern  = `\s`
+	spacesReplacer = "-"
+
+	invalidCharacters   = `[^a-z0-9.:_-]`
+	invalidCharReplacer = ""
+)
+
+var (
+	// IsValidServiceKey verifies if the service key is a valid one.
+	// A valid service key is something like 'service_token:service_name'.
+	// The service_token should be of 64 characters long and the size of
+	// service_name is larger than 0 but up to 255 characters.
+	IsValidServiceKey = regexp.MustCompile(validServiceKeyPattern).MatchString
 
-// IsValidServiceKey verifies if the service key is a valid one.
-// A valid service key is something like 'service_token:service_name'.
-// The service_token should be of 64 characters long and the size of
-// service_name is larger than 0 but up to 255 characters.
-var IsValidServiceKey = regexp.MustCompile(validServiceKeyPattern).MatchString
+	// ReplaceSpacesWith replaces all the spaces with valid characters (hyphen)
+	ReplaceSpacesWith = regexp.MustCompile(spacesPattern).ReplaceAllString
 
-// ToServiceKey converts a string to a service key
+	// RemoveInvalidChars remove invalid characters
+	RemoveInvalidChars = regexp.MustCompile(invalidCharacters).ReplaceAllString
+)
+
+// ToServiceKey converts a string to a service key. The argument should be
+// a valid service key string.
+//
+// It doesn't touch the service key but does the following to the original
+// service name:
+// - convert all characters to lowercase
+// - convert spaces to hyphens
+// - remove invalid characters ( [^a-z0-9.:_-])
 func ToServiceKey(s string) interface{} {
-	return s
+	keyS := strings.Split(s, serviceKeyDelimiter)
+	if len(keyS) != serviceKeyPartsCnt {
```

Comment:
What if the service name itself contains `:`? Think that's a valid character and this logic might mistakenly thinks that's not a valid service key? ie len(keyS) would be > 2

Reply:
Hmm it makes sense, thanks.